requirements.txt

```diff
@@ -31,7 +31,7 @@ setuptools>=75.0
 # pip - MIT license
 pip>=22.0
 # build - MIT license
-build>=1.1.1
+build>=1.1.1, !=1.2.2.post1
```

Comment:
@coderabbitai,
CI found found a test bug:

> E AssertionError: Regex didn't match: '^\s*[a-zA-Z0-9_\-\.]+(?:,?\s?(?:==|!=|>=|<=|>|<)\s?[0-9\.]+)+(?:\s*(?:#.*)?)$' not found in 'build>=1.2.1, !=1.2.2.post1' : Invalid requirement format at line 69: 'build>=1.2.1, !=1.2.2.post1'

this is coming from the test code in `tests/test_deps.py` around line 69.

The regular expression misses PEP-440 style `a#`, `b#`, `rc#`, `.pre#` and `.post#` suffixes.

Help me fix the regular expression.